system_baseline/validators.py

```diff
@@ -0,0 +1,47 @@
+from uuid import UUID
+from http import HTTPStatus
+
+from system_baseline.exceptions import FactValidationError, HTTPError
+
+FACTS_MAXSIZE = 2 ** 19  # 512KB
+
+
+def check_for_duplicate_names(facts):
+    """
+    check if any names are duplicated; raises an exception if duplicates are found.
+    """
+    names = []
+    for fact in facts:
+        if "value" in fact:
+            names.append(fact["name"])
+        elif "values" in fact:
+            check_for_duplicate_names(fact["values"])
+
+    for name in names:
+        if names.count(name) > 1:
+            raise FactValidationError("name %s declared more than once" % name)
+
+
+def check_facts_length(facts):
+    """
+    check if fact length is greater than FACTS_MAXSIZE
+    """
+    if len(str(facts)) > FACTS_MAXSIZE:
+        raise FactValidationError(
+            "attempted to save fact list over %s bytes" % FACTS_MAXSIZE
+        )
+
+
+def validate_uuids(baseline_ids):
+    """
+    helper method to test if a UUID is properly formatted. Will raise an
+    exception if format is wrong.
+    """
+    for baseline_id in baseline_ids:
+        try:
+            UUID(baseline_id)
+        except ValueError:
+            raise HTTPError(
+                HTTPStatus.BAD_REQUEST,
+                message="baseline_id %s is not a UUID" % baseline_id,
+            )
```

Comment:
This file contains two naming conventions for validator functions. Might be nice to unify the naming scheme. They also do two different error raising: some raise ValidationErrors to allow the view layer to handle the HTTP coding, and in one case the validator raises an HTTP error code directly. I tend to like the former model better with a refactor like this.

Reply:
I agree. Will update